skiplang/prelude/runtime/palloc.c

```diff
@@ -587,23 +587,33 @@ int sk_is_static(void* ptr) {
 /* Free table. */
 /*****************************************************************************/
 
-size_t sk_bit_size(size_t size) {
-  return (size_t)(sizeof(size_t) * 8 - __builtin_clzl(size - 1));
+typedef size_t slot_t;
+
+#if !(defined(__has_builtin) && __has_builtin(__builtin_stdc_bit_width))
+static inline size_t __builtin_stdc_bit_width(size_t size) {
+  return size ? (size_t)(sizeof(size_t) * 8UL - __builtin_clzl(size)) : 0;
+}
+#endif
+
+slot_t sk_slot_of_size(size_t size) {
+  // Must return a value between 0 and FTABLE_SIZE - 1 included
+  if (__builtin_expect(size < sizeof(void*), 0)) {
+    size = sizeof(void*);
+  }
+  return __builtin_stdc_bit_width(size - 1);
 }
 
-size_t sk_pow2_size(size_t size) {
-  size = (size + (sizeof(void*) - 1)) & ~(sizeof(void*) - 1);
-  return (1 << sk_bit_size(size));
+size_t sk_size_of_slot(slot_t slot) {
+  // Must return a multiple of sizeof(void*) and at least sizeof(void*)
```

Comment:
This is a bit confusing. Maybe explicitly state that `slot` is expected to be greater or equal to `log2(sizeof(void *))`?